course_info/static/course_info/js/controllers/PeopleController.js

```diff
@@ -120,49 +120,76 @@
                 .finally($scope.updateProgressBar);
         };
         $scope.addPeopleToCourse = function(searchTerms) {
+
             /* looks up HUIDs, XIDs, and/or email addresses from searchTerms
              and attempts to add people to the course who do not already have an
              enrollment.
              */
-            var membersByUserId = {};
-            $scope.clearMessages();
-            $scope.operationInProgress = true;
-            var searchTermList = $scope.getSearchTermList(searchTerms);
-            $scope.tracking.total = searchTermList.length;
-            $scope.updateProgressBar('Looking up ' + $scope.tracking.total
-                + ' people');
-            var memberPromise = $scope.lookupCourseMembers()
-                .then(function updateCourseMembers(memberResponse) {
-                    membersByUserId = $scope.getMembersByUserId(
-                        memberResponse.data.results);
-                    return memberResponse;
-                }, function courseMemberLookupFailed(memberResponse) {
-                    $scope.handleAjaxErrorResponse(memberResponse);
-                    $scope.messages.warnings.push({
-                        type: 'courseMemberLookupFailed'
+
+            // split the list to get the number of people the user is trying to add
+            $scope.numPeople = searchTerms.split(',').length;
+            $scope.selectedRoleName = $scope.selectedRole.roleName;
+            // open a modal confirmation box and as the user to verify they want to add
+            // the number of users they entered.
+            var modalInstance = $uibModal.open({
+                animation: true,
+                templateUrl: 'partials/add-people-to-course-confirmation.html',
+                controller: function ($scope, $uibModalInstance, numPeople, selectedRoleName) {
+                    $scope.numPeople = numPeople;
+                    $scope.selectedRoleName = selectedRoleName;
+                },
+                resolve: {
+                    numPeople: function () {
+                        return $scope.numPeople;
+                    },
+                    selectedRoleName: function () {
+                        return $scope.selectedRoleName;
+                    }
+                }
+            });
+
+            modalInstance.result.then(function modalSuccess() {
```

Comment:
To avoid further indentation/nesting, we can just call a modified version of the original `AddPeopleToCourse()` and put all the modal stuff in its own scope-level function.